context/rootfs/scripts/init-registry.sh

```diff
@@ -29,34 +30,56 @@ certs_dir="$rootfs/certs"
 
 mkdir -p "$VOLUME" || true
 
+imageServer="nerdctl"
+if [[ $(ls ../cri/docker*.tar.gz) ]]; then
+    imageServer="docker"
+fi
+
 startRegistry() {
   n=1
   while ((n <= 3)); do
     echo "attempt to start registry"
-    nerdctl start $container && break
-    ((n++))
+    $imageServer start $container; break || ((n++))
```

Comment:
I am wondering if it is quite hard to understand. Please split them into single lines.